workspace/TS100/Core/BSP/Miniware/flash.c

```diff
@@ -33,7 +33,7 @@ uint8_t flash_save_buffer(const uint8_t *buffer, const uint16_t length) {
 	HAL_FLASH_Unlock();
 	for (uint8_t i = 0; i < (length / 2); i++) {
 		resetWatchdog();
-		HAL_FLASH_Program(FLASH_TYPEPROGRAM_HALFWORD, FLASH_ADDR + (i * 2),
+		HAL_FLASH_Program(FLASH_TYPEPROGRAM_HALFWORD, (uint32_t)&settings_page[i],

```

Comment:
Does this work -> Does the compiler automatically convert this to a uint16_t read?

Reply:
In  this specific case no reads are happening, just address calculation. And since settings_page is declared as an array of uint16_t, pointer arithmetics does its job as expected. 16-bit reads would work too.

Reply:
Ah yes, my bad for mis-reading this